HenBot/SettingsHandler.cs

```diff
@@ -38,30 +25,27 @@ public static class SettingsHandler
         switch (savedUser.Step)
         {
             case 0:
-                await ProcessStep0(botClient, update, chatId, savedUser, cancellationToken);
+                await ProcessStep0(botClient, update, chatId, cancellationToken);
                 break;
             case 1:
-                await ProcessStep1(botClient, update, chatId, savedUser, cancellationToken);
-                break;
-            case 2:
-                await ProcessStep2(botClient, update, chatId, savedUser, cancellationToken);
+                await ProcessStep1(botClient, update, chatId, cancellationToken);
                 break;
         }
     }
 
     private static async Task ProcessStep0(ITelegramBotClient botClient, Update update, long chatId,
-        SavedUser savedUser,
         CancellationToken cancellationToken)
     {
         if (int.TryParse(update.Message.Text, out var limit) && limit <= 10)
         {
-            savedUser.Limit = limit;
-            savedUser.Step++;
+            userToSave.Limit = limit;
+            var user = UserRepository.GetUser(chatId);
+            user.Step++;
+            UserRepository.UpdateUser(user);
 
             await botClient.SendTextMessageAsync(
                 chatId,
-                "Ok now choose a rating.",
-                replyMarkup: inlineKeyboard,
+                "Ok now write search queries.",
```

Comment:
Give the user an example of search queries or/and a pattern, not everyone knows how to use gelbooru